eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/EwonTagQuality.java

```diff
@@ -0,0 +1,307 @@
+   *
+   * @param quality the Ewon tag quality string
+   * @throws IllegalArgumentException if the specified quality string is {@code null} or invalid
+   * @since 1.0.0
+   */
+  public EwonTagQuality(String quality) {
+    // Convert quality string to integer
+    int qualityInt;
+    if (quality == null) {
+      throw new IllegalArgumentException("Quality cannot be null");
+    } else if (quality.equalsIgnoreCase(DMWEB_QUALITY_GOOD_STRING)) {
+      qualityInt = DMWEB_QUALITY_GOOD;
+    } else if (quality.equalsIgnoreCase(DMWEB_QUALITY_BAD_STRING)) {
+      qualityInt = DMWEB_QUALITY_BAD;
+    } else if (quality.equalsIgnoreCase(DMWEB_QUALITY_UNCERTAIN_STRING)) {
+      qualityInt = DMWEB_QUALITY_UNCERTAIN;
+    } else if (quality.equalsIgnoreCase(DMWEB_QUALITY_INITIAL_GOOD_STRING)) {
+      qualityInt = DMWEB_QUALITY_INITIAL_GOOD;
+    } else if (quality.equalsIgnoreCase(DMWEB_QUALITY_UNKNOWN_STRING)) {
+      qualityInt = DMWEB_QUALITY_UNKNOWN;
+    } else {
+      throw new IllegalArgumentException("Invalid quality string");
+    }
+
+    // Set quality integer
+    this.quality = qualityInt;
+  }
+
+  /**
+   * Returns the Ewon tag quality.
+   *
+   * @return the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getQuality() {
+    return quality;
+  }
+
+  /**
+   * Returns the vendor-specific quality information of the Ewon tag quality.
+   *
+   * @return the vendor-specific quality information of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getVendorSpecificQualityInformation() {
+    return (quality >> 8) & 0xFF;
+  }
+
+  /**
+   * Returns the major quality of the Ewon tag quality.
+   *
+   * @return the major quality of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getMajorQuality() {
+    return (quality >> 6) & 0x03;
+  }
+
+  /**
+   * Returns the sub-status of the Ewon tag quality.
+   *
+   * @return the sub-status of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getSubStatus() {
+    return (quality >> 2) & 0x0F;
+  }
+
+  /**
+   * Returns the limit status of the Ewon tag quality.
+   *
+   * @return the limit status of the Ewon tag quality
+   * @since 1.0.0
+   */
+  public int getLimitStatus() {
+    return quality & 0x03;
+  }
+
+  /**
+   * Returns the Ewon tag quality as an Ignition {@link QualityCode} object.
+   *
+   * @return the Ewon tag quality as an Ignition {@link QualityCode} object
+   * @since 1.0.0
+   */
+  public QualityCode getQualityCode() {
+    QualityCode result;
+    switch (getMajorQuality()) {
+      case 0:
+        result = QualityCode.Bad;
+        break;
+      case 1:
+        result = QualityCode.Uncertain;
+        break;
+      case 3:
+        result = QualityCode.Good;
+        break;
+      default:
+        result = QualityCode.Bad_OutOfRange;
+        break;
+    }
+
+    // If quality is not good, check sub-status
+    if (result != QualityCode.Good) {
```

Comment:
The returned QualityCode variable will be overwritten if there is an applicable sub-status. Otherwise, it returns whatever QualityCode was selected above using the major quality.